File: zatca_erpgulf/zatca_erpgulf/country.py
"""Country normalization shared by ZATCA validation and document generators."""
from __future__ import annotations

from typing import Any

from .country_code import country_code_mapping


_SAUDI_ALIASES = {"sa", "s.a.", "saudi arabia", "kingdom of saudi arabia"}


def normalize_country_code(value: Any) -> str:
    """Return an ISO-3166 alpha-2 code without changing stored ERPNext data.

    ERPNext stores Country links by name on common versions, while integrations
    and imports may provide an ISO code. Both forms are accepted.
    """
    normalized = str(value or "").strip().casefold()
    if not normalized:
        return ""
    if normalized in _SAUDI_ALIASES:
        return "SA"
    if len(normalized) == 2 and normalized.isalpha():
        return normalized.upper()
    return country_code_mapping().get(normalized, normalized.upper())


def is_saudi_country(value: Any) -> bool:
    return normalize_country_code(value) == "SA"

File: zatca_erpgulf/public/js/country.js
// Shared country normalization for ZATCA client-side checks.
// Accepts ERPNext Country names and ISO-like values without changing stored data.
window.zatca_normalize_country_code = window.zatca_normalize_country_code || function (value) {
    const normalized = String(value || "").trim().toLowerCase();
    if (!normalized) return "";
    if (["sa", "s.a.", "saudi arabia", "kingdom of saudi arabia"].includes(normalized)) return "SA";
    if (/^[a-z]{2}$/.test(normalized)) return normalized.toUpperCase();
    return normalized;
};

window.zatca_is_saudi_country = window.zatca_is_saudi_country || function (value) {
    return window.zatca_normalize_country_code(value) === "SA";
};
